### PR TITLE
feat: webwallet and governance framework integration

### LINKED ISSUE #320: WebWallet : Governance Framework Integration

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- closes #320

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: cmd/user-agent/src/pages/chapi/Governance.vue [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+                        <div class="md-subhead">{{govnVC.credentialSubject.description}}</div>
+                    </md-card-header-text>
+
+                    <md-card-media>
+                        <img :src="govnLogoSrc" @error="imageError = true">
+                    </md-card-media>
+                </md-card-header>
+
+                <md-subheader>
+                    <span style="margin-top: 10px"  v-if="issuer">As per <b style="font-size: 16px">{{govnVC.credentialSubject.name}}</b> rules, issuers should prove
```
Comment (troyronda) [inline cmd/user-agent/src/pages/chapi/Governance.vue:45]: this text should be updated.

File: cmd/user-agent/src/pages/chapi/wallet/get/getCredentialsByQuery.js [hunk 2]
```diff
@@ -36,6 +36,15 @@ export class WalletGetByQuery extends WalletGet {
...
         this.invitation = jp.query(credEvent, '$..credentialRequestOptions.web.VerifiablePresentation.query[?(@.type=="DIDConnect")].invitation');
 
+        /*
+          TODO:
+           * current assumption - expecting only one governance VC in request, may be support for multiple
+           * correlate governance VC with requesting party so that consent for trust gets shown only once
+           * verify governance VC proof
+           * verify requesting party in governance framework to make sure this party of behaving properly
+         */
+        this.govnVC = jp.query(credEvent, '$..credentialRequestOptions.web.VerifiablePresentation.query[?(@.type=="DIDConnect")].credentials[?(@.type[0]=="GovernanceCredential" || @.type[1]=="GovernanceCredential")]');
```
Comment (troyronda) [inline cmd/user-agent/src/pages/chapi/wallet/get/getCredentialsByQuery.js:46]: where does the issuer / verifier get challenged to produce a VP of the governance credential?

**Comment by troyronda** [comment]
Note: This change isn’t yet a governance framework integration, but rather an expectation of a credential schema held by agencies making requests to the wallet.

File: cmd/user-agent/src/pages/webWalletSamples.js [hunk 3]
```diff
@@ -99,6 +105,41 @@ const udc = {
...
+        "geos": ["Canadian"],
+        "jurisdictions": ["ca"],
+        "logo": "https://example.com/logo",
+        "name": "Trustbloc Govn",
+        "privileges": [{"name": "accredit", "uri": "https://example.com/accredit"}],
+        "roles": ["accreditor"],
+        "topics": ["banking"],
+        "version": "1.0"
+    },
+    "issuer": "did:trustbloc:testnet.trustbloc.local:EiDdRGN4x2S4D0xYTPaJEHcD50Sq5fgv0sUfbdgY7x6lkQ",
```
Comment (troyronda) [inline cmd/user-agent/src/pages/webWalletSamples.js:132]: In a followup, we will need to be more sophisticated about which issuers and credentials are acceptable for this purpose.

File: cmd/user-agent/src/pages/webWalletSamples.js [hunk 4]
```diff
@@ -99,6 +105,41 @@ const udc = {
     }
 }
 
+const govnVC = {
```
Comment (troyronda) [inline cmd/user-agent/src/pages/webWalletSamples.js:108]: note: this is a governance agency credential, not the credential that the issuer/verifier would be holding.

File: cmd/user-agent/src/pages/chapi/Governance.vue [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+                        <div class="md-subhead">This {{ issuer ? 'issuer' : 'verifier'}} may not be trustworthy</div>
+                    </md-card-header-text>
+
+                    <md-card-media>
+                        <img :src="govnLogoSrc" @error="imageError = true">
+                    </md-card-media>
+                </md-card-header>
+
+                <md-subheader>
+                    <span style="margin-top: 10px" v-if="issuer">Issuers should prove they're accredited by trusted governance authority before issuing credentials. The current issuer,
```
Comment (troyronda) [inline cmd/user-agent/src/pages/chapi/Governance.vue:113]: this text should be updated.

File: cmd/user-agent/src/pages/chapi/DIDConnect.vue [hunk 6]
```diff
@@ -30,17 +30,17 @@ SPDX-License-Identifier: Apache-2.0
                     </div>
 
                     <md-card-content class="viewport">
-                        <p class="viewport"> Hello {{ this.walletUser.id }},</p>
-                            <span style="color: #0E9A00">{{ requestOrigin }}</span> would like to
+                        Below issuer would like to
```
Comment (troyronda) [inline cmd/user-agent/src/pages/chapi/DIDConnect.vue:33]: we probably shouldn’t start sentences with Below.

Comment (troyronda) [reply cmd/user-agent/src/pages/chapi/DIDConnect.vue:33]: also is this specific to issuers?

**Comment by troyronda** [comment]
I wonder if we need a feature flag - this feature is incomplete.

File: cmd/user-agent/src/pages/chapi/DIDConnect.vue (hunk 6 above)
Comment (sudeshrshetty) [reply cmd/user-agent/src/pages/chapi/DIDConnect.vue:33]: Fixed the text, will push a new PR.

For now, yes this DIDconnect happens only with issuers, RP sends DIDconnect request as part of presentation exchange query, which is handled by different vue file.

File: cmd/user-agent/src/pages/chapi/Governance.vue (hunk 5 above)
Comment (sudeshrshetty) [reply cmd/user-agent/src/pages/chapi/Governance.vue:113]: any suggested texts?

File: cmd/user-agent/src/pages/webWalletSamples.js (hunk 3 above)
Comment (sudeshrshetty) [reply cmd/user-agent/src/pages/webWalletSamples.js:132]: Will add it in follow up issue.

File: cmd/user-agent/src/pages/chapi/wallet/get/getCredentialsByQuery.js (hunk 2 above)
Comment (sudeshrshetty) [reply cmd/user-agent/src/pages/chapi/wallet/get/getCredentialsByQuery.js:46]: updated TODO in next PR, and will be added in follow up issue.

File: cmd/user-agent/src/pages/chapi/Governance.vue (hunk 1 above)
Comment (sudeshrshetty) [reply cmd/user-agent/src/pages/chapi/Governance.vue:45]: Will add it in follow up issue.

**Pull request merged**